x-pack/legacy/plugins/siem/public/components/timeline/query_bar/index.tsx

```diff
@@ -0,0 +1,309 @@
+              const filterWithoutDropArea = filterManager
+                .getFilters()
+                .filter((f: Filter) => f.meta.controlledBy !== timelineFilterDropArea);
+              setFilters(filterWithoutDropArea);
+              setQueryBarFilters(filterWithoutDropArea);
+            }
+          },
+        })
+      );
+
+      return () => {
+        isSubscribed = false;
+        subscriptions.unsubscribe();
+      };
+    }, []);
+
+    useEffect(() => {
+      const filterWithoutDropArea = filterManager
+        .getFilters()
+        .filter((f: Filter) => f.meta.controlledBy !== timelineFilterDropArea);
+      if (!isEqual(filters, filterWithoutDropArea)) {
+        filterManager.setFilters(filters);
+      }
+    }, [filters]);
+
+    useEffect(() => {
+      setFilterQueryConverted({
+        query: filterQuery != null ? filterQuery.expression : '',
+        language: filterQuery != null ? filterQuery.kind : 'kuery',
+      });
+    }, [filterQuery]);
+
+    useEffect(() => {
+      setDataProvidersDsl(
+        convertKueryToElasticSearchQuery(
+          buildGlobalQuery(dataProviders, browserFields),
+          indexPattern
+        )
+      );
+    }, [dataProviders, browserFields, indexPattern]);
+
+    useEffect(() => {
+      if (fromStr != null && toStr != null) {
+        setDateRangeFrom(fromStr);
+        setDateRangTo(toStr);
+      } else if (from != null && to != null) {
+        setDateRangeFrom(new Date(from).toISOString());
+        setDateRangTo(new Date(to).toISOString());
+      }
+    }, [from, fromStr, to, toStr]);
+
+    useEffect(() => {
+      let isSubscribed = true;
+      async function setSavedQueryByServices() {
+        if (savedQueryId != null && savedQueryServices != null) {
+          const mySavedQuery = await savedQueryServices.getSavedQuery(savedQueryId);
+          if (isSubscribed) {
+            setSavedQuery({
+              ...mySavedQuery,
+              attributes: {
+                ...mySavedQuery.attributes,
+                filters: filters.filter(f => f.meta.controlledBy !== timelineFilterDropArea),
+              },
+            });
+          }
+        } else if (isSubscribed) {
+          setSavedQuery(null);
+        }
+      }
+      setSavedQueryByServices();
+      return () => {
+        isSubscribed = false;
+      };
+    }, [savedQueryId]);
+
+    const onChangedQuery = useCallback(
+      (newQuery: Query) => {
+        if (
+          filterQueryDraft == null ||
+          ((filterQueryDraft != null && filterQueryDraft.expression !== newQuery.query) ||
+            filterQueryDraft.kind !== newQuery.language)
+        ) {
+          setKqlFilterQueryDraft(
+            newQuery.query as string,
+            newQuery.language as KueryFilterQueryKind
+          );
+        }
+      },
+      [filterQueryDraft]
+    );
+
+    const onSubmitQuery = useCallback(
+      (newQuery: Query, timefilter?: SavedQueryTimeFilter) => {
+        // TODO need to add time
```

Comment:
Does `time` need to be added as part of this PR, or part of a future feature?

Reply:
I should have removed this comment, I did add it [here](https://github.com/elastic/kibana/pull/49813/files/77b85f5f82e2922216011d04d1b74efa728fa327#diff-d7cc2c225d2f65deccd6c39b200d2bb7R213)